server/libs/domain/src/share/share.core.ts

```diff
@@ -63,13 +63,24 @@ export class ShareCore {
     return this.repository.remove(link);
   }
 
-  async updateAssets(userId: string, id: string, assets: AssetEntity[]) {
+  async addAssets(userId: string, id: string, assets: AssetEntity[]) {
     const link = await this.get(userId, id);
     if (!link) {
       throw new BadRequestException('Shared link not found');
     }
 
-    return this.repository.save({ ...link, assets });
+    return this.repository.save({ ...link, assets: [...link.assets, ...assets] });
```

Comment:
Will this cause an error if an asset already had been added to the shared link? I assume there is a unique constant they will be violated. How is that handled in the add to album flow?

Reply:
They just got added to the album and the shared link. There is no error regarding the unique constraint.